tests/cluster/tests/04-resharding.tcl

```diff
@@ -140,7 +140,12 @@ test "Verify $numkeys keys for consistency with logical content" {
 test "Terminate and restart all the instances" {
     foreach_redis_id id {
         # Stop AOF so that an initial AOFRW won't prevent the instance from terminating
-        R $id config set appendonly no
+        # Retrying in a loop, slave might reply -LOADING if it is flushing or loading db.
+        wait_for_condition 50 1000 {
+            [R $id config set appendonly no] eq {OK}
```

Comment:
maybe we should make an effort to allow this config change during loading?
it's certainly possible.. when turning it on, we just need to remember to start aof when loading is over (which we do anyway), and turning it off is a common procedure for a forceful shutdown (before we had SHUTDOWN FORCE)

Reply:
I've made the change to allow appendonly config change during loading. As you said, we are already turning it on/off while loading rdb on replication. Also, added code to apply config change lazily if it is changed inside loadDataFromDisk() on startup.